src/Agent.Worker/TaskRunner.cs

```diff
@@ -17,13 +17,35 @@ public interface ITaskRunner : IStep, IAgentService
 
     public sealed class TaskRunner : AgentService, ITaskRunner
     {
-        public bool AlwaysRun => TaskInstance?.AlwaysRun ?? default(bool);
+        public string Condition
+        {
+            get
+            {
+                if (ExecutionContext.Features.HasFlag(PlanFeatures.TaskCondition) || !string.IsNullOrEmpty(ExecutionContext.Variables.Get("VSTS_TEMP_FEATURE_TASKCONDITION")))
```

Comment:
The VSTS_TEMP___ variables are temporary until web API and server side changes are in place.